sys/fmt/fmt.c

```diff
@@ -243,6 +243,56 @@ size_t fmt_s16_dfp(char *out, int16_t val, unsigned fp_digits)
     return pos;
 }
 
+static const uint32_t _tenmap[] = {
+    0,
+    10LU,
+    100LU,
+    1000LU,
+    10000LU,
+    100000LU,
+    1000000LU,
+    10000000LU,
+};
+
+/* this is very probably not the most efficient implementation, as it at least
+ * pulls in floating point math.  But it works, and it's always nice to have
+ * low hanging fruits when optimizing. (Kaspar)
+ */
+size_t fmt_float(char *out, float f, unsigned precision)
+{
+    assert (precision <= 7);
+
+    unsigned negative = (f < 0);
+    uint32_t integer;
+
+    if (negative) {
+        f *= -1;
+    }
+
+    integer = (uint32_t) f;
+    f -= integer;
+
+    uint32_t fraction = f * _tenmap[precision];
+
+    size_t res = negative;
+    if (negative && out) {
+        *out++ = '-';
+    }
+
+    res += fmt_u32_dec(out, integer);
+    if (precision && fraction) {
+        if (out) {
+            out += res;
+            *out++ = '.';
+            size_t tmp = fmt_u32_dec(out, fraction);
+            fmt_lpad(out, tmp, precision, '0');
+        }
+        res += (1 + precision);
```

Comment:
Now I saw that `fmt_lpad` documentation is wrong (didn't catch it before). It says: "returns pad_len" but the first lines of the method already contradict that... I created an issue for that #6806